tests/loaddata/test_digits_access.py

```diff
@@ -51,3 +53,38 @@ def test_get_train_test(dataset_name, download_path):
     assert source.n_classes() == 10
     assert isinstance(source_train, torch.utils.data.Dataset)
     assert isinstance(source_test, torch.utils.data.Dataset)
+
+
+@pytest.mark.parametrize("dataset_name", ALL)
+@pytest.mark.parametrize("val_ratio", VAL_RATIO)
+@pytest.mark.parametrize("class_sub_sample", CLASS_SUB_SAMPLES)
+def test_class_subsampling(dataset_name, download_path, val_ratio, class_sub_sample):
+    source, target, num_channels = DigitDataset.get_source_target(
+        DigitDataset(dataset_name), DigitDataset(dataset_name), download_path
+    )
+
+    source_train = source.get_class_subsampled_train(class_sub_sample)
+    source_test = source.get_class_subsampled_test(class_sub_sample)
+    source_train_val = source.get_train_val(val_ratio, class_sub_sample)
+
+    dataset = MultiDomainDatasets(source, target, config_weight_type=WEIGHT_TYPE[0], config_size_type=DATASIZE_TYPE[1])
+    dataset.prepare_data_loaders()
+    dataset_subsampled = MultiDomainDatasets(
+        source,
+        target,
+        config_weight_type=WEIGHT_TYPE[0],
+        config_size_type=DATASIZE_TYPE[1],
+        sub_class_ids=class_sub_sample,
+    )
+    dataset_subsampled.prepare_data_loaders()
+
+    assert len(dataset_subsampled) <= len(dataset)
+    assert len(source_train) <= len(source.get_train())
+    assert len(source_test) <= len(source.get_test())
+
+    assert isinstance(source_train, torch.utils.data.Dataset)
```

Comment:
These assertions are less useful, right? We used similar assertions when we do not have stronger assertions implementation. In your case, the above can be stronger assertions. Consider to remove these.

Reply:
Agree with Haiping. In addition, my most assertions are too general and simple and only check some parameter type, because we want to achieve high coverage in the first step. However, from now on, we are going to improve our test example by replacing some assertions with stronger ones. Therefore, in your test example, you can explore assertions for a more detailed test than the current code.